api/src/main/java/uk/gov/dhsc/htbhf/claimant/repository/ClaimantRepository.java

```diff
@@ -5,17 +5,45 @@
 import org.springframework.data.repository.query.Param;
 import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
 
+import java.util.Optional;
 import java.util.UUID;
 
 /**
  * JPA repository for the Claimant table.
  */
 public interface ClaimantRepository extends CrudRepository<Claimant, UUID> {
 
-    @Query("SELECT COUNT(claimant) FROM Claimant claimant WHERE claimant.nino = :nino AND claimant.eligibilityStatus = 'ELIGIBLE'")
-    Long getNumberOfMatchingEligibleClaimants(@Param("nino") String nino);
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.nino = :nino "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithNino(@Param("nino") String nino);
 
-    default Boolean eligibleClaimExists(String nino) {
-        return getNumberOfMatchingEligibleClaimants(nino) != 0;
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.dwpHouseholdIdentifier = :dwpHouseholdIdentifier "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithDwpHouseholdIdentifier(@Param("dwpHouseholdIdentifier") String dwpHouseholdIdentifier);
+
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.hmrcHouseholdIdentifier = :hmrcHouseholdIdentifier "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithHmrcHouseholdIdentifier(@Param("hmrcHouseholdIdentifier") String hmrcHouseholdIdentifier);
+
+    default boolean eligibleClaimExistsForNino(String nino) {
+        return countEligibleClaimantsWithNino(nino) != 0;
+    }
+
+    default boolean eligibleClaimExistsForHousehold(Optional<String> dwpHouseholdIdentifier, Optional<String> hmrcHouseholdIdentifier) {
```

Comment:
I don't think this kind of logic belongs in the repository. I'm also not a fan of passing optionals into methods, I'd have the checks done in the calling class.

Reply:
This has been moved out of the repository and optionals removed